js/components/multi-input.js

```diff
@@ -0,0 +1,88 @@
+define([
+  'knockout',
+  'text!./multi-input.html',
+	'utils/CommonUtils',
+  'components/Component',
+  'utils/AutoBind',
+  'lodash',  
+  'databindings',
+  'less!./multi-input.less',
+], function (
+  ko, 
+    view,
+    commonUtils,
+    Component,
+    AutoBind,
+    _,
+) {
+    const typeIdentifier = {
+      integer: "integer",
+      float: "float"
+    }
+    class MultiInput extends AutoBind(Component) {
+      constructor(params) {
+        super(params);
+        this.selectedValueType = params.selectedValueType || "integer";
+        this.selectedValueExtender = params.selectedValueExtender || {};
```

Comment:
Shouldn't the `selectedValueExtender` be picked based on `selectedValueType` to have a single source of truth? E.g. now it is possible to pass selectedValueType = 'integer' and no extender or extender for float

Reply:
I see your point @pavgra - let me explain a bit more why I did it this way and perhaps there is another way to approach it. The `selectedValueExtender` is applied to the `itemToAdd` which has the effect of enforcing the extender rules for the value you are entering in the text box. For integers/floats, the `numeric` extender does a pretty good job of enforcing a standard. I also wanted to have a way to also enforce some type of validation before you add the item to the list. So, I added the `selectedValueType` to add in the proper parser to and to use that before adding the item to the list.

If I understand your suggestion, the extender and type should be embedded into the component so that it is simply a matter of specifying the `selectedValueType` only. If so, I agree and I'll make that change.

Reply:
> so that it is simply a matter of specifying the `selectedValueType` only. If so, I agree and I'll make that change.

yes, to avoid cases when you provide selectedValueType = 'integer' but the component allows string to be entered